packages/esm-patient-labs-app/src/lab-orders/add-lab-order/lab-order-form.component.tsx

```diff
@@ -22,32 +49,38 @@ export function LabOrderForm({ initialOrder, closeWorkspace }: LabOrderFormProps
   const isTablet = useLayoutType() === 'tablet';
   const session = useSession();
   const { orders, setOrders } = useOrderBasket<LabOrderBasketItem>('labs', prepLabOrderPostData);
-  const [inProgressLabOrder, setInProgressLabOrder] = useState(initialOrder as LabOrderBasketItem);
   const { testTypes, isLoading: isLoadingTestTypes, error: errorLoadingTestTypes } = useTestTypes();
 
+  const { control, handleSubmit, getValues, formState } = useForm<LabOrderBasketItem>({
+    mode: 'all',
+    resolver: zodResolver(labOrderFormSchema),
+    defaultValues: {
+      ...initialOrder,
+      instructions: '',
+      labReferenceNumber: '',
+    },
+  });
```

Comment:
```suggestion
  const { control, handleSubmit, getValues, formState: {isDirty} } = useForm<LabOrderBasketItem>({
    mode: 'all',
    resolver: zodResolver(labOrderFormSchema),
    defaultValues: {
      ...initialOrder,
      instructions: '',
      labReferenceNumber: '',
    },
  });
  
  useEffect(() => {
  	promptBeforeClosing(() => isDirty);
  }, [isDirty])
```

Reply:
@vasharma05 to add `promptBeforeClosing`, has the related PR been merged yet,
Thanks

Reply:
This functionality is already present. My PR adds it in the workspace.

This workspace will otherwise miss out on that (parallel PRs)

Reply:
I have added it, however looks like implementation of `promptBeforeClosing` in this PR is different from your PR. This requires to pass `workspacename` as arg